sklearn/metrics/classification.py

```diff
@@ -1434,6 +1438,37 @@ class 2       1.00      0.67      0.80         3
                              np.sum(s),
                              width=width, digits=digits)
 
+    if output_dict is not False:
+
+        headers = headers[1:]
+
+        scores = [p,r,f1,s]
```

Comment:
Then you can do `averages = np.average(scores, axis=0, weights=s)`. But apart from averages, to make dicts of dicts, can do something like:
```python
out = dict(zip(headers, [dict(zip(target_names, x)) for x in scores]))
```